app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/LeaveEventReviewFragment.kt

```diff
@@ -0,0 +1,157 @@
+import com.github.sdpteam15.polyevents.model.observable.Observable
+
+/**
+ * A Dialog Fragment that is displayed over an EventActivity, to leave a review for the event.
+ */
+class LeaveEventReviewFragment(val eventId: String?):
+    DialogFragment(R.layout.fragment_leave_review) {
+
+    private lateinit var leaveReviewDialogConfirmButton: Button
+    private lateinit var userFeedbackDialogEditText: EditText
+    private lateinit var leaveReviewDialogRatingBar: RatingBar
+    private lateinit var leaveReviewDialogCancelButton: Button
+
+    private var rated = false
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState)
+        if (view != null) {
+            leaveReviewDialogConfirmButton = view.findViewById(R.id.leave_review_fragment_save_button)
+
+            userFeedbackDialogEditText = view.findViewById(R.id.leave_review_fragment_feedback_text)
+
+            leaveReviewDialogRatingBar = view.findViewById(R.id.leave_review_fragment_rating)
+            leaveReviewDialogRatingBar.setOnRatingBarChangeListener { _, _, _ ->
+                rated = true
+            }
+
+            leaveReviewDialogCancelButton = view.findViewById(R.id.leave_review_fragment_cancel_button)
+            leaveReviewDialogCancelButton.setOnClickListener {
+                // Dimiss the dialog if canceled
+                dismiss()
+            }
+
+            leaveReviewDialogConfirmButton.setOnClickListener {
+                onClickAdd()
+            }
+
+            val ratingObservable = Observable<Rating>()
+            ratingObservable.observe(this) {
+                Log.d(TAG, "Retrieved Rating for $eventId and ${currentDatabase.currentUser!!.uid}!")
+                val rating = it.value
+                leaveReviewDialogRatingBar.rating = rating.rate!!
+                if (rating.feedback != null) {
+                    if (!rating.feedback.isEmpty()) {
+                        userFeedbackDialogEditText.setText(rating.feedback)
+                    }
+                }
+                leaveReviewDialogConfirmButton.setOnClickListener {
+                    onClickUpdate(rating)
+                }
+            }
+
+            currentDatabase.eventDatabase!!.getUserRatingFromEvent(
+                userId = currentDatabase.currentUser!!.uid,
+                eventId = eventId!!,
+                returnedRating = ratingObservable,
+                userAccess = null
+            )
+        }
+
+        return view
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // Uncomment the following lines to manually set the height and the width of the DialogFragment
+        /*val width = ConstraintLayout.LayoutParams.MATCH_PARENT
+        val height = ConstraintLayout.LayoutParams.MATCH_PARENT
+        dialog!!.window!!.setLayout(width, height)*/
```

Comment:
Should we keep these lines if they are commented ?

Reply:
I kept them in case I decide to implement another dialogFragment, it's to remember to use these if I need to set the width and height of the dialog programmatically.